### PR TITLE
TXMLEngine: buffer expand error

### PR DESCRIPTION
**Pull request by linev**
In TXMLEngine in several places pointer was not recalculated after expand of the buffer.
Problem exists since the beginning (approx 2004), but never appeared before.
Most probably while realloc() function in previous glibc versions was always returning same pointer value. Now it is not a case. Reported here: https://root-forum.cern.ch/t/29593

Same patch can be applied to all previous ROOT branches.

P.S. cmake /path/to/roottest does not work for me

### CODE REVIEW

**Comment by linev** [comment]
@phsft-bot build

File: io/xml/src/TXMLEngine.cxx [hunk 1]
```diff
@@ -286,19 +286,20 @@ class TXMLInputStream {
    ////////////////////////////////////////////////////////////////////////////
    /// Allocate more data for the buffer, preserving content
 
-   Bool_t ExpandStream()
+   Bool_t ExpandStream(char **curr = nullptr)
```
Comment (Axel-Naumann) [inline io/xml/src/TXMLEngine.cxx]: All but one call to `ExpandStream` pass `curr`. Can you just force `ExpandStream(char* &curr)`?

Comment (linev) [reply io/xml/src/TXMLEngine.cxx]: But at that place I do not have any extra pointer which should be recalculated.
Shoud I create dummy for that?

Comment (Axel-Naumann) [reply io/xml/src/TXMLEngine.cxx]: In my personal preference, yes - I hate `**`! :-) But your decision, I was merely suggesting this (in that single case, you could move the declaration of `curr` up, before the call to `ExpandStream(curr)`). Let me know what you prefer and I'll merge.

**Comment by linev** [comment]
@Axel-Naumann 
I adjust the code - now it is reference and I even can avoid use of dummy there.

I still cannot run roottest standalone.

**Comment by Axel-Naumann** [comment]
@amadio could maybe look at the standalone roottest issue - is that known?

We check out roottest into src/ and configure ROOT with `-Dtesting=On -Droottest=On` and it picks up roottest and enabled `ctest`.

But Jenkins ran roottest for us and is happy :-)

**Comment by Axel-Naumann** [comment]
@pcanal please consider for post-merge review whenever you're available.

**Comment by linev** [comment]
@Axel-Naumann 
This patch should be applied to all relevant branches - 6-14, 6-12, maybe 6-10
It fixes real problem for all components, where TXMLEngine is used - like GDML import, TMVA gui and so on.

**Comment by amadio** [comment]
@Axel-Naumann There is a PR for this, but waiting on comments from me to be addressed. If it's urgent, I can fix it.

**Comment by amadio** [comment]
Hi Sergei, this is known. It's issue https://sft.its.cern.ch/jira/browse/ROOT-9405. The PR with a fix for it is here https://github.com/root-project/roottest/pull/192, but I asked for changes, and they were not made yet. If it's too annoying for you let me know and I'll apply a fix myself. That's what I meant before. Cheers,

**Pull request merged**